tests/integration/pilot/workloadentry_test.go

```diff
@@ -0,0 +1,157 @@
+	"istio.io/istio/pkg/test/framework/components/echo/check"
+	commonDeployment "istio.io/istio/pkg/test/framework/components/echo/common/deployment"
+	"istio.io/istio/pkg/test/framework/components/echo/common/ports"
+	"istio.io/istio/pkg/test/framework/components/echo/deployment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/resource/config/apply"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+func TestWorkloadEntry(t *testing.T) {
+	// nolint: staticcheck
+	framework.NewTest(t).
+		RequiresMinClusters(2).
+		Features("traffic.reachability").
+		Run(func(t framework.TestContext) {
+			ist, err := istio.Get(t)
+			if err != nil {
+				t.Fatal(err)
+			}
+			clusterCfg := t.Clusters().Default()
+			namespaceName := apps.Namespace.Name()
+
+			// Create a echo deployment "z" in the echos namespace.
+			t.Logf("Deploy an echo instance in namespace %s on cluster %s", namespaceName, clusterCfg.Name())
+			deployment.New(t, clusterCfg).
+				WithConfig(echo.Config{
+					Service:        "z",
+					ServiceAccount: true,
+					Namespace:      apps.Namespace,
+					Ports:          ports.All(),
+					Subsets:        []echo.SubsetConfig{{}},
+				}).BuildOrFail(t)
+
+			// Define an AUTO_PASSTHROUGH EW gateway
+			gatewayCfg := `apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: ingress-ew
+  namespace: istio-system
+spec:
+  selector:
+    istio: eastwestgateway
+  servers:
+  - port:
+      number: 15443
+      name: https
+      protocol: TLS
+    hosts:
+    - serviceentry.mesh.global
+    tls:
+      mode: AUTO_PASSTHROUGH
+`
+			// Configure an AUTO_PASSTHROUGH EW gateway
+			t.ConfigIstio().YAML("istio-system", gatewayCfg).ApplyOrFail(t, apply.CleanupConditionally)
+
+			ewGatewayIP, ewGatewayPort := ist.EastWestGatewayFor(clusterCfg).AddressForPort(15443)
+			if ewGatewayIP == "" || ewGatewayPort == 0 { // most likely EW gateway is not deployed, skip testing
+				t.Skipf("Skipping test, eastwest gateway is probably not deployed for cluster %s", clusterCfg.Name())
+			}
+
+			// Define ServiceEntry & WorkloadEntry for test
+			workloadEntryYaml := fmt.Sprintf(`apiVersion: networking.istio.io/v1beta1
+kind: ServiceEntry
+metadata:
+  name: z-se
+spec:
+  addresses:
+  - 240.240.34.56
+  hosts:
+  - serviceentry.mesh.global
+  ports:
+  - name: http
+    number: 80
+    protocol: HTTP
+    targetPort: 8080
+  location: MESH_INTERNAL
+  resolution: STATIC
+  workloadSelector:
+    labels:
+      app: z
+---
+apiVersion: networking.istio.io/v1beta1
+kind: WorkloadEntry
+metadata:
+  name: z-we
+  labels:
+    security.istio.io/tlsMode: istio
+    app: z
+spec:
+  ports:
+    http: %v
+  address: %s`, ewGatewayPort, ewGatewayIP)
+
+			// Configure ServiceEntry & WorkloadEntry
+			t.ConfigIstio().YAML(namespaceName, workloadEntryYaml).ApplyOrFail(t, apply.CleanupConditionally)
+
+			srcs := apps.All.Instances()
+			for _, src := range srcs {
```

Comment:
There are some idiomatic parts of the `echotest` package that can abstract out this loop and skip the "special" workloads (so if more are added in the future, you don't have to append to this skip list)

Reply:
Thanks for the suggestion. Will take a look and revise.